### PR TITLE
Update "Running BOUT++" documentation

### LINKED ISSUE #2467: Misleading documentation about `archive`/restart files
Reported from slack:

> According to the user doc https://bout-dev.readthedocs.io/en/latest/user_docs/running_bout.html "Archived restart files have names like “BOUT.restart_0020.#.nc”, and are written every 20 outputs by default. To change this, set “archive” in the BOUT.inp file."

And reply from @johnomotani:

> It looks like the manual is out of date (sorry about that!). The archive option was removed in this commit e35f1f761ffa98911fd3efbf398ee4e00dcd4166.
I'd recommend using xBOUT (https://github.com/boutproject/xBOUT) to create restart files from the output data. There is a to_restart() method for this https://xbout.readthedocs.io/en/latest/xbout.html#xbout.boutdataset.BoutDatasetAccessor.to_restart. Not sure if it's obvious how to use, so please feel free to message again! The xBOUT docs are a bit sparse - this is making me think that we could use an FAQ section to include common recipes like 'how to make restart files'...

### PR DESCRIPTION
**Pull request by ZedThree**
Quite a bit of the manual section on running BOUT++ was out of date. There's certainly still lots more we should rewrite, but this fixes some glaring errors at least.

Fixes #2467

### CODE REVIEW

File: manual/sphinx/user_docs/running_bout.rst [hunk 1]
```diff
@@ -301,153 +245,149 @@ over this see if there are any important warnings or errors. Each
...
-            First       :  Fourth order central (C4)
-            Second      :  Fourth order central (C4)
-            Upwind      :  Third order WENO (W3)
-            Flux        :  Split into upwind and central (SPLIT)
-    Setting Z differencing methods
-            First       :  FFT (FFT)
-            Second      :  FFT (FFT)
-            Upwind      :  Third order WENO (W3)
-            Flux        :  Split into upwind and central (SPLIT)
-
```
Comment (ZedThree) [inline manual/sphinx/user_docs/running_bout.rst:375]: It is a bit of a shame to have lost this output. Should we reinstate it?

Comment (johnomotani) [reply manual/sphinx/user_docs/running_bout.rst:375]: :+1: I agree would be nice to reinstate it.

File: manual/sphinx/user_docs/running_bout.rst [hunk 2]
```diff
@@ -301,153 +245,149 @@ over this see if there are any important warnings or errors. Each
...
+
+This will always be processor number ’0’ on screen as only the output
+from processor ’0’ is sent to the terminal.
+
+The process ID (pid) is also printed::
+
+    pid: 17835
+
+which is useful for distinguishing multiple simulations running at the
+same time and, for example, stop one run if it starts misbehaving.
```
Comment (johnomotani) [inline manual/sphinx/user_docs/running_bout.rst:279]:
```suggestion
same time and, for example, to stop one run if it starts misbehaving.
```

**Review by johnomotani — APPROVED** [review]
This is a definite improvement! :+1:

**Review by johnomotani — APPROVED** [review]

**Pull request merged**